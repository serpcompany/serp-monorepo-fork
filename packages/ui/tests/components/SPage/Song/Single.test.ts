import { mockNuxtImport } from '@nuxt/test-utils/runtime';
import { describe, expect, it } from 'vitest';
import SPageSongSingle from '../../../../components/SPage/Song/Single.vue';
import ComponentRender from '../../../componentRender';
import '../../../mockUseUserSession';

mockNuxtImport('useSeoMeta', () => () => {});

let config_: Record<string, unknown> = {
  app: { baseURL: '/' },
  public: {
    useAuth: false
  }
};
let songData_: unknown = {};
let upvotesData_: unknown = { upvotes: [] };

mockNuxtImport('useRuntimeConfig', () => () => config_);
mockNuxtImport('useSong', () => () => songData_);
mockNuxtImport('useFetchWithCache', () => () => upvotesData_);

const scenarios: [
  string,
  { config: Record<string, unknown>; song: unknown; upvotes: string[] }
][] = [
  [
    'complete song with upvotes and auth enabled',
    {
      config: { app: { baseURL: '/' }, public: { useAuth: true } },
      song: {
        name: 'Test Song',
        slug: 'test-song',
        overview: '<p>Overview content for Test Song.</p>',
        genres: ['Rock', 'Pop'],
        tags: ['Live', 'Studio'],
        lyrics: '<p>Test lyrics content.</p>',
        releaseGroups: [
          {
            slug: 'album-1',
            cover_art_urls: { '500': '/album-1.jpg' },
            name: 'Album One',
            recordings: [
              {
                slug: 'song-1',
                has_lyrics: true,
                name: 'Song One',
                position: 1
              },
              {
                slug: 'song-2',
                has_lyrics: false,
                name: 'Song Two',
                position: 2
              }
            ]
          }
        ],
        artists: [
          { slug: 'artist-1', credit_name: 'Artist One', join_phrase: '' }
        ]
      },
      upvotes: ['user@example.com']
    }
  ],
  [
    'minimal song without upvotes and auth disabled',
    {
      config: { app: { baseURL: '/' }, public: { useAuth: false } },
      song: {
        name: 'Minimal Song',
        slug: 'minimal-song',
        overview: '',
        genres: [],
        tags: [],
        releaseGroups: [],
        artists: []
      },
      upvotes: []
    }
  ]
];

describe('SPageSongSingle Snapshot', () => {
  it.each(scenarios)(
    'renders %s correctly',
    async (desc: string, { config, song, upvotes }) => {
      config_ = config;
      songData_ = song;
      upvotesData_ = { upvotes };

      const html = await ComponentRender(
        `SPageSongSingle - ${desc}`,
        {},
        SPageSongSingle
      );
      expect(html).toMatchSnapshot();
    }
  );
});
