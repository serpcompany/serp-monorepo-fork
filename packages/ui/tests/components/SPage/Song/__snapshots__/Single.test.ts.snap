// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SPageSongSingle Snapshot > renders complete song with upvotes and auth enabled correctly 1`] = `
"<div class="flex flex-col lg:grid lg:grid-cols-10 lg:gap-10">
  <!--v-if-->
  <div class="lg:col-span-10">
    <!---->
    <header class="bg-background sticky top-0 z-40 border-b border-gray-200 transition-all duration-300 dark:border-gray-800 bg-background sticky top-0 z-10 transition-all duration-300">
      <div class="w-full max-w-(--ui-container) mx-auto px-4 sm:px-6 lg:px-8">
        <div class="flex flex-col items-center justify-between gap-4 py-3 sm:flex-row">
          <div class="flex min-w-0 flex-grow items-center gap-3">
            <!--v-if-->
            <div class="min-w-0 flex-grow">
              <h1 class="truncate text-xl font-bold text-gray-900 transition-all duration-300 sm:text-2xl dark:text-white">Test Song by Artist One</h1>
              <div class="truncate text-sm text-gray-500 dark:text-gray-400"></div>
            </div>
          </div>
          <div class="flex w-full flex-shrink-0 items-center justify-end gap-3 sm:w-auto">
            <div module="music_songs">
              <div class="relative inline-flex -space-x-px"><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-up shrink-0 size-5" aria-hidden="true"></span>
                  <!--v-if-->
                  <!--v-if-->
                </button><span class="font-medium inline-flex items-center not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-xs gap-1 rounded-md ring ring-inset ring-accented text-default bg-default p-1 w-8 justify-center"><!--v-if--><span class="truncate">NaN</span>
                <!--v-if--></span><button type="button" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors not-only:first:rounded-e-none not-only:last:rounded-s-none not-last:not-first:rounded-none focus-visible:z-[1] text-sm gap-1.5 ring ring-inset ring-accented text-default bg-default hover:bg-elevated disabled:bg-default aria-disabled:bg-default focus:outline-none focus-visible:ring-2 focus-visible:ring-inverted p-1.5"><span class="iconify i-lucide:arrow-big-down shrink-0 size-5" aria-hidden="true"></span>
                  <!--v-if-->
                  <!--v-if-->
                </button>
              </div>
            </div><a href="https://serp.ly/@daftfm/amazon/music/unlimited" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 w-full sm:w-auto"><span class="iconify i-heroicons:arrow-top-right-on-square shrink-0 size-4" aria-hidden="true"></span><span class="truncate">Listen Free</span>
              <!--v-if-->
            </a>
          </div>
        </div>
        <nav class="-mx-4 flex overflow-x-auto border-t border-gray-200 px-4 sm:-mx-6 sm:px-6 dark:border-gray-800"><a href="#lyrics" class="font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 flex-shrink-0 rounded-none whitespace-nowrap">
            <!--v-if--><span class="truncate">Lyrics</span>
            <!--v-if-->
          </a></nav>
      </div>
    </header><!-- Main content with grid -->
    <div class="mx-auto max-w-7xl p-4 md:p-6 lg:p-8">
      <!-- Breadcrumbs -->
      <nav aria-label="breadcrumb" class="relative min-w-0 mb-6">
        <ol class="flex items-center gap-1.5"></ol>
      </nav><!-- Main Content (70) -->
      <div class="grid gap-6 lg:grid-cols-3">
        <div class="lg:col-span-2">
          <!-- Song content -->
          <div id="lyrics" data-adsense-content="song-lyrics">
            <h2 class="mb-4 text-xl font-semibold">Test Song Lyrics</h2><!-- eslint-disable-next-line vue/no-v-html -->
            <div class="prose dark:prose-invert max-w-none whitespace-pre-line">
              <p>Test lyrics content.</p>
            </div>
          </div>
        </div><!-- Sidebar (30%) -->
        <div class="space-y-6 lg:col-span-1">
          <!-- Add Album Cover Art -->
          <!--v-if-->
          <!-- Song Info Section -->
          <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
            <div class="p-4 sm:px-6">
              <div class="font-medium">Song Info</div>
            </div>
            <div class="p-4 sm:p-6">
              <div class="space-y-2">
                <div class="mb-2"><span class="font-medium">Artist: </span><a href="/artists/artist-1" class="text-primary hover:underline">Artist One</a></div>
                <!--v-if-->
                <div><span class="font-medium">Genres:</span> Rock, Pop</div>
                <div><span class="font-medium">Tags:</span> Live, Studio</div>
                <div class="pt-2">&lt;p&gt;Overview content for Test Song.&lt;/p&gt;</div>
              </div>
            </div>
            <!--v-if-->
          </div><!-- Album Songs Section -->
          <!--v-if-->
          <!-- Artist's Other Albums Section -->
          <!--v-if-->
          <!-- Listen Section -->
          <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
            <div class="p-4 sm:px-6">
              <div class="font-medium">Listen on</div>
            </div>
            <div class="p-4 sm:p-6">
              <div class="p-1"><a href="https://serp.ly/@daftfm/amazon/music/unlimited" rel="noopener noreferrer" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-3 py-2 text-sm gap-2 w-full justify-center text-primary focus:outline-none focus-visible:bg-primary/15 disabled:bg-primary/10 aria-disabled:bg-primary/10 bg-blue-50 hover:bg-blue-100 dark:bg-blue-900/20 dark:hover:bg-blue-900/30">
                  <!--v-if--> Amazon Music
                  <!--v-if-->
                </a></div>
            </div>
            <!--v-if-->
          </div>
        </div>
      </div>
    </div>
  </div>
  <!--v-if-->
</div>"
`;

exports[`SPageSongSingle Snapshot > renders minimal song without upvotes and auth disabled correctly 1`] = `
"<div class="flex flex-col lg:grid lg:grid-cols-10 lg:gap-10">
  <!--v-if-->
  <div class="lg:col-span-10">
    <!---->
    <header class="bg-background sticky top-0 z-40 border-b border-gray-200 transition-all duration-300 dark:border-gray-800 bg-background sticky top-0 z-10 transition-all duration-300">
      <div class="w-full max-w-(--ui-container) mx-auto px-4 sm:px-6 lg:px-8">
        <div class="flex flex-col items-center justify-between gap-4 py-3 sm:flex-row">
          <div class="flex min-w-0 flex-grow items-center gap-3">
            <!--v-if-->
            <div class="min-w-0 flex-grow">
              <h1 class="truncate text-xl font-bold text-gray-900 transition-all duration-300 sm:text-2xl dark:text-white">Minimal Song by Unknown Artist</h1>
              <div class="truncate text-sm text-gray-500 dark:text-gray-400"></div>
            </div>
          </div>
          <div class="flex w-full flex-shrink-0 items-center justify-end gap-3 sm:w-auto"><a href="https://serp.ly/@daftfm/amazon/music/unlimited" rel="noopener noreferrer" target="_blank" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 w-full sm:w-auto"><span class="iconify i-heroicons:arrow-top-right-on-square shrink-0 size-4" aria-hidden="true"></span><span class="truncate">Listen Free</span>
              <!--v-if-->
            </a></div>
        </div>
        <nav class="-mx-4 flex overflow-x-auto border-t border-gray-200 px-4 sm:-mx-6 sm:px-6 dark:border-gray-800"><a href="#lyrics" class="font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-2.5 py-1.5 text-xs gap-1.5 flex-shrink-0 rounded-none whitespace-nowrap">
            <!--v-if--><span class="truncate">Lyrics</span>
            <!--v-if-->
          </a></nav>
      </div>
    </header><!-- Main content with grid -->
    <div class="mx-auto max-w-7xl p-4 md:p-6 lg:p-8">
      <!-- Breadcrumbs -->
      <nav aria-label="breadcrumb" class="relative min-w-0 mb-6">
        <ol class="flex items-center gap-1.5"></ol>
      </nav><!-- Main Content (70) -->
      <div class="grid gap-6 lg:grid-cols-3">
        <div class="lg:col-span-2">
          <!-- Song content -->
          <div id="lyrics" data-adsense-content="song-lyrics">
            <h2 class="mb-4 text-xl font-semibold">Minimal Song Lyrics</h2><!-- eslint-disable-next-line vue/no-v-html -->
            <div class="prose dark:prose-invert max-w-none whitespace-pre-line"></div>
          </div>
        </div><!-- Sidebar (30%) -->
        <div class="space-y-6 lg:col-span-1">
          <!-- Add Album Cover Art -->
          <!--v-if-->
          <!-- Song Info Section -->
          <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
            <div class="p-4 sm:px-6">
              <div class="font-medium">Song Info</div>
            </div>
            <div class="p-4 sm:p-6">
              <div class="space-y-2">
                <!--v-if-->
                <!--v-if-->
                <!--v-if-->
                <!--v-if-->
                <!--v-if-->
              </div>
            </div>
            <!--v-if-->
          </div><!-- Album Songs Section -->
          <!--v-if-->
          <!-- Artist's Other Albums Section -->
          <!--v-if-->
          <!-- Listen Section -->
          <div class="rounded-lg bg-default ring ring-default divide-y divide-default">
            <div class="p-4 sm:px-6">
              <div class="font-medium">Listen on</div>
            </div>
            <div class="p-4 sm:p-6">
              <div class="p-1"><a href="https://serp.ly/@daftfm/amazon/music/unlimited" rel="noopener noreferrer" class="rounded-md font-medium inline-flex items-center disabled:cursor-not-allowed aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 transition-colors px-3 py-2 text-sm gap-2 w-full justify-center text-primary focus:outline-none focus-visible:bg-primary/15 disabled:bg-primary/10 aria-disabled:bg-primary/10 bg-blue-50 hover:bg-blue-100 dark:bg-blue-900/20 dark:hover:bg-blue-900/30">
                  <!--v-if--> Amazon Music
                  <!--v-if-->
                </a></div>
            </div>
            <!--v-if-->
          </div>
        </div>
      </div>
    </div>
  </div>
  <!--v-if-->
</div>"
`;
